pkg/controller/util/planner/planner.go

```diff
@@ -71,25 +72,41 @@ func (p *Planner) Plan(availableClusters []string, currentReplicaCount map[strin
 	plan := make(map[string]int64, len(preferences))
 	overflow := make(map[string]int64, len(preferences))
 
-	named := func(name string, pref fedschedulingv1a1.ClusterPreferences) *namedClusterPreferences {
+	named := func(name string, pref fedschedulingv1a1.ClusterPreferences) (*namedClusterPreferences, error) {
 		// Seems to work better than addler for our case.
 		hasher := fnv.New32()
-		hasher.Write([]byte(name))
-		hasher.Write([]byte(replicaSetKey))
+		if _, err := hasher.Write([]byte(name)); err != nil {
+			return nil, err
+		}
+		if _, err := hasher.Write([]byte(replicaSetKey)); err != nil {
+			return nil, err
+		}
 
 		return &namedClusterPreferences{
 			clusterName:        name,
 			hash:               hasher.Sum32(),
 			ClusterPreferences: pref,
-		}
+		}, nil
 	}
 
 	for _, cluster := range availableClusters {
 		if localRSP, found := p.preferences.Spec.Clusters[cluster]; found {
-			preferences = append(preferences, named(cluster, localRSP))
+			preference, err := named(cluster, localRSP)
+			if err != nil {
+				glog.Errorf("Unexpected err: %v\n", err)
```

Comment:
Looking at this in a wider context this function should return the error rather than logging and continuing.  My apologies for not figuring this out sooner.

cc: @irfanurrehman

Reply:
Please take a look at the latest commit, it returns errors in both method - `plan` and `schedule`